### PR TITLE
feat: Added support for cloud init

### LINKED ISSUE #457: Add support for custom cloud-init for node pools
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

OKE now supports custom cloud-init: https://blogs.oracle.com/cloud-infrastructure/post/container-engine-for-kubernetes-custom-worker-node-startup-script-support

It would be great if we can have support for passing the new cloud init as a parameter like we can for compute instances.

### Potential Terraform Configuration

```hcl
# Copy-paste any Terraform configurations for how the requested feature may be used. 
```

### PR DESCRIPTION
**Pull request by hyder**
Closes #457 and #414. Added support for cloud-init. Used cloud-init script to run oci-growfs to expand boot volume size. Updated docs and added subsections for bastion, operator instructions to make
table of content render better.

### CODE REVIEW

File: modules/oke/cloudinit.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
+locals {
+  worker_script_template = templatefile("${path.module}/cloudinit/worker.template.sh",
+    {
+      worker_timezone = "Australia/Sydney"
```
Comment (karthicgit) [inline modules/oke/cloudinit.tf:4]: worker_timezone is hardcoded.

File: modules/oke/cloudinit.tf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  }
+  
+  # example for adding more script
+  # part {
+  #   filename     = "second.sh"
+  #   content_type = "text/x-shellscript"
+  #   content      = local.second_script_template
+  # }
+}
+
```
Comment (karthicgit) [inline modules/oke/cloudinit.tf:29]: may i know how the cloud init scripts are received from the end user when they are using oke module.They are not supposed to edit any tf files.

**Review by karthicgit — COMMENTED** [review]
Reviewed

File: modules/oke/cloudinit.tf (hunk 2 above)
Comment (hyder) [reply modules/oke/cloudinit.tf:29]: They have to add their scripts in the cloudinit directory, load them as templates in locals, and then edit the cloudinit_config data source in cloudinit.tf. In this way, it is an explicit action on the user's part to load a customer script whereas the worker.template.sh, they are not supposed to modify.

**Review by karthicgit — APPROVED** [review]

**Pull request merged**